rand_core/src/lib.rs

```diff
@@ -397,54 +456,56 @@ pub trait SeedableRng: Sized {
 // Implement `RngCore` for references to an `RngCore`.
 // Force inlining all functions, so that it is up to the `RngCore`
 // implementation and the optimizer to decide on inlining.
-impl<'a, R: RngCore + ?Sized> RngCore for &'a mut R {
-    #[inline(always)]
-    fn next_u32(&mut self) -> u32 {
-        (**self).next_u32()
-    }
+impl<'a, R: TryRngCore + ?Sized> TryRngCore for &'a mut R {
```

Comment:
Unfortunately, even though in theory this blanket impl together with impl for `R: TryRngCore<Error = Infallible>` effectively combines into `impl<R: RngCore> RngCore for &mut R { ... }`. But this reasoning is only valid assuming that `RngCore` is not implemented directly. We can not enforce it at compile time, so compiler can not take it into account.